sdks/python/apache_beam/runners/worker/sideinputs.py

```diff
@@ -68,9 +73,36 @@ def __init__(self, sources,
     # Whether an error was encountered in any source reader.
     self.has_errored = False
 
+    self.read_counter = read_counter or opcounters.TransformIOCounter()
+
     self.reader_threads = []
     self._start_reader_threads()
 
+  def add_byte_counter(self, reader):
+    """Adds byte counter observer to a side input reader.
+
+    If the 'sideinput_io_metrics' experiment flag is not passed in, then nothing
+    is attached to the reader.
+
+    Args:
+      reader: A reader that should inherit from ObservableMixin to have
+        bytes tracked.
+    """
+
+    # The tracking of time spend reading and bytes read from side inputs is kept
+    # behind an experiment flag to test performance impact.
+    experiments = RuntimeValueProvider('experiments', str, '').get().split(',')
+    if 'sideinput_io_metrics' not in experiments:
+      return
```

Comment:
<!--thread_id:cc_163113306_t; commit:ee3c5cfd0adc6bf2848f2010c0beec4d9ace5123; resolved:1-->
<!--section:context-quote-->
> **charlesccychen** wrote:
> Can you do this at the beginning of `_reader_thread` so that we do the check of the RuntimeValueProvider exactly once?

<!--section:body-->
Done.